packages/teams-js/src/private/messageChannels.ts

```diff
@@ -0,0 +1,77 @@
+import { requestPortFromParentWithVersion } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag, getLogger } from '../internal/telemetry';
+import { runtime } from '../public/runtime';
+
+/**
+ * @hidden
+ * Namespace to request message ports from the host application.
+ *
+ * @beta
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export namespace messageChannels {
+  let telemetryPort: MessagePort | undefined;
+
+  const messageChannelsTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_1;
+
+  const logger = getLogger('messageChannels');
+  /**
+   * @hidden
+   * @beta
+   *
+   * Fetches a MessagePort to batch telemetry through the host's telemetry worker.
+   * The port is cached once received, so subsequent calls return the same port.
+   * @returns MessagePort.
+   *
+   * @throws Error if {@linkcode app.initialize} has not successfully completed,
+   * if the host does not support the feature, or if the port request is rejected.
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export async function getTelemetryPort(): Promise<MessagePort> {
+    // If the port has already been initialized, return it.
+    if (telemetryPort) {
+      logger('Returning telemetry port from cache');
+      return telemetryPort;
+    }
+
+    isSupported();
```

Comment:
```suggestion
          if (!isSupported()) {
        throw errorNotSupportedOnPlatform;
      }
```

Reply:
Oops I had that before, and obviously messed it up in refactoring! Good catch.